features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/transport/IpFixMessageBuilder.java

```diff
@@ -283,52 +513,50 @@ public IpFixMessageBuilder(Iterable<Value<?>> values, RecordEnrichment enrichmen
+                        double interval = samplingFlowInterval != null ?
+                                          samplingFlowInterval.doubleValue() : 1.0;
+                        double spacing = samplingFlowSpacing != null ?
+                                         samplingFlowSpacing.doubleValue() : 0.0;
+                        double samplingIntervalValue = interval + spacing / interval;
                         builder.setSamplingInterval(setDoubleValue(samplingIntervalValue));
                         break;
-                    case 2:
-                        Double flowSamplingTimeInterval = this.flowSamplingTimeInterval != null ?
-                                this.flowSamplingTimeInterval.doubleValue() : 1.0;
-                        Double flowSamplingTimeSpacing = this.flowSamplingTimeSpacing != null ?
-                                this.flowSamplingTimeSpacing.doubleValue() : 0.0;
-                        samplingIntervalValue = flowSamplingTimeInterval + flowSamplingTimeSpacing / flowSamplingTimeSpacing;
+                    }
+                    case 2: {
+                        double interval = flowSamplingTimeInterval != null ?
+                                          flowSamplingTimeInterval.doubleValue() : 1.0;
+                        double spacing = flowSamplingTimeSpacing != null ?
+                                         flowSamplingTimeSpacing.doubleValue() : 0.0;
+                        double samplingIntervalValue = interval + spacing / spacing;
                         builder.setSamplingInterval(setDoubleValue(samplingIntervalValue));
                         break;
-                    case 3:
-                        Double samplingSize = this.samplingSize != null ?
-                                this.samplingSize.doubleValue() : 1.0;
-                        Double samplingPopulation = this.samplingPopulation != null ?
-                                this.samplingPopulation.doubleValue() : 1.0;
-                        samplingIntervalValue = samplingPopulation / samplingSize;
+                    }
+                    case 3: {
+                        double size = samplingSize != null ? samplingSize.doubleValue() : 1.0;
+                        double population = samplingPopulation != null ? samplingPopulation.doubleValue() : 1.0;
+                        double samplingIntervalValue = population / size;
                         builder.setSamplingInterval(setDoubleValue(samplingIntervalValue));
                         break;
-                    case 4:
-                        Double samplingProbability = this.samplingProbability != null ?
-                                this.samplingProbability.doubleValue() : 1.0;
-                        builder.setSamplingInterval(setDoubleValue(1.0 / samplingProbability));
+                    }
+                    case 4: {
+                        Double probability = samplingProbability != null ? samplingProbability.doubleValue() : 1.0;
```

Comment:
Any reason to have a Double instead of a double here?

Reply:
I've just moved that code and didn't bother to make it nice :). The whole section needs reworking, TBH.